common/models/user.js

```diff
@@ -870,6 +870,16 @@ module.exports = function(User) {
     next();
   });
 
+  User.observe('before save', function emptyPasswordUpdate(ctx, next) {
+    if ((ctx.instance && ctx.instance.password === '') || (ctx.data && ctx.data.password === '')) {
```

Comment:
So far, the password validation is happening in `User.validatePassword`, see https://github.com/strongloop/loopback/blob/c7cd43c90ad4f8e52d70289468264a35a3df0efe/common/models/user.js#L646

Is there any particular reason why this new check cannot be performed by that method too?